llvm/lib/Target/RISCV/RISCVExpandPseudoInsts.cpp

```diff
@@ -538,14 +540,48 @@ bool RISCVExpandPseudo::expandAuipccInstPair(
 
   MF->insert(++MBB.getIterator(), NewMBB);
 
-  BuildMI(NewMBB, DL, TII->get(RISCV::AUIPCC), TmpReg)
-      .addDisp(Symbol, 0, FlagsHi);
+  auto CheriotCapImportAttrName =
+      llvm::cheri::CapabilityImportAttr::getAttrName();
+  std::optional<Attribute> CheriotCapImportAttr = std::nullopt;
+
+  if (Symbol.isGlobal()) {
+    if (auto *GV = llvm::dyn_cast<llvm::GlobalVariable>(Symbol.getGlobal())) {
```

Comment:
Reduce nesting like so:

```
auto *GV = ...
if (GV && GV->hasAttribute(...)) {
```

Reply:
https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/be0f7cd362897c448d0d4aa8693ef3effdfde30a reduces the nesting.